infra/feast-operator/internal/controller/featurestore_controller_test.go

```diff
@@ -1130,3 +1223,25 @@ func getFeatureStoreYamlEnvVar(envs []corev1.EnvVar) *corev1.EnvVar {
 	}
 	return nil
 }
+
+func areEnvVarArraysEqual(arr1 []corev1.EnvVar, arr2 []corev1.EnvVar) bool {
+	if len(arr1) != len(arr2) {
+		return false
+	}
+
+	// Create a map to count occurrences of EnvVars in the first array.
+	envMap := make(map[string]string)
+
+	for _, env := range arr1 {
+		envMap[env.Name] = env.Value
+	}
+
+	// Check the second array against the map.
+	for _, env := range arr2 {
+		if val, exists := envMap[env.Name]; !exists || val != env.Value {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
i wonder if `reflect.DeepEqual()` might be a better approach? instead of this `areEnvVarArraysEqual` function?

Reply:
> i wonder if `reflect.DeepEqual()` might be a better approach? instead of this `areEnvVarArraysEqual` function?

It doesn't take into account 2 arrays are the same if they have the same elements but different order:

```package main

import (
	"fmt"
	"reflect"
)

func main() {
	arr1 := [3]int{1, 2, 3}
	arr2 := [3]int{1, 2, 3}
	arr3 := [3]int{3, 2, 1}

	fmt.Println(reflect.DeepEqual(arr1, arr2)) // Output: true
	fmt.Println(reflect.DeepEqual(arr1, arr3)) // Output: false
```

Reply:
the main problem i see with the `areEnvVarArraysEqual` function is its not checking `env.ValueFrom`. maybe we use `reflect.DeepEqual(env, env)` at the EnvValue level somehow... within this custom function?

Reply:
Done

Reply:
maybe `DeepEqual` isn't even needed? `envMap[env.Name] == env` ??